server.go

```diff
@@ -2,64 +2,75 @@ package main
+	"github.com/ipfs/boxo/routing/http/client"
+	"github.com/ipfs/boxo/routing/http/contentrouter"
+	"github.com/ipfs/boxo/routing/http/server"
+	"github.com/ipfs/boxo/routing/http/types"
+	"github.com/ipfs/boxo/routing/http/types/iter"
 	"github.com/ipfs/go-cid"
-	"github.com/ipfs/go-ipns"
 	"github.com/libp2p/go-libp2p"
-	"github.com/libp2p/go-libp2p-core/host"
-	"github.com/libp2p/go-libp2p-core/peer"
-	"github.com/libp2p/go-libp2p-core/routing"
 	dht "github.com/libp2p/go-libp2p-kad-dht"
 	"github.com/libp2p/go-libp2p-kad-dht/fullrt"
 	record "github.com/libp2p/go-libp2p-record"
-	rhelpers "github.com/libp2p/go-libp2p-routing-helpers"
-
-	drc "github.com/ipfs/go-delegated-routing/client"
-	drp "github.com/ipfs/go-delegated-routing/gen/proto"
-	drs "github.com/ipfs/go-delegated-routing/server"
+	routinghelpers "github.com/libp2p/go-libp2p-routing-helpers"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/routing"
+	rcmgr "github.com/libp2p/go-libp2p/p2p/host/resource-manager"
 )
 
-func start(ctx context.Context, port int, runAcceleratedDHTClient bool) error {
-	indexerClient, err := drp.New_DelegatedRouting_Client(devIndexerEndpoint)
-	if err != nil {
-		return err
-	}
-
+func start(ctx context.Context, port int, runAcceleratedDHTClient bool, contentEndpoints, peerEndpoints, ipnsEndpoints []string) error {
 	h, err := newHost(runAcceleratedDHTClient)
 	if err != nil {
 		return err
 	}
 
-	var d routing.Routing
-
+	var dhtRouting routing.Routing
 	if runAcceleratedDHTClient {
-		wrappedDHT, err := wrapAcceleratedAndRegularClient(ctx, h)
+		wrappedDHT, err := newWrappedStandardAndAcceleratedDHTClient(ctx, h)
 		if err != nil {
 			return err
 		}
-		d = wrappedDHT
+		dhtRouting = wrappedDHT
 	} else {
-		bsPeers, err := peer.AddrInfosFromP2pAddrs(dht.DefaultBootstrapPeers...)
+		standardDHT, err := dht.New(ctx, h, dht.Mode(dht.ModeClient), dht.BootstrapPeers(dht.GetDefaultBootstrapPeerAddrInfos()...))
 		if err != nil {
 			return err
 		}
-		standardDHT, err := dht.New(ctx, h, dht.Mode(dht.ModeClient), dht.BootstrapPeers(bsPeers...))
-		if err != nil {
-			return err
-		}
-		d = standardDHT
+		dhtRouting = standardDHT
+	}
+
+	crRouters, err := getCombinedRouting(contentEndpoints, dhtRouting)
+	if err != nil {
+		return err
+	}
+
+	prRouters, err := getCombinedRouting(peerEndpoints, dhtRouting)
+	if err != nil {
+		return err
+	}
+
+	ipnsRouters, err := getCombinedRouting(ipnsEndpoints, dhtRouting)
+	if err != nil {
+		return err
 	}
 
 	proxy := &delegatedRoutingProxy{
-		indexer: &contentRoutingIndexerWrapper{indexer: drc.NewContentRoutingClient(drc.NewClient(indexerClient))},
-		dht:     d,
+		cr: crRouters,
+		pr: prRouters,
+		vs: ipnsRouters,
 	}
 
-	f := drs.DelegatedRoutingAsyncHandler(proxy)
-	http.Handle("/", f)
+	log.Printf("Listening on http://0.0.0.0:%d", port)
+	log.Printf("Delegated Routing API on http://127.0.0.1:%d/routing/v1", port)
```

Comment:
Might be fine, but also a little confusing since routing/v1 is on 127.0.0.1 and all other interfaces too.